apps/synapse-portal-framework/tsconfig.json

```diff
@@ -1,17 +1,39 @@
 {
   "extends": "../../shared/tsconfig.base.json",
   "compilerOptions": {
+    "composite": true,
     "rootDir": ".",
     "baseUrl": ".",
+    "outDir": "../../node_modules/.cache/.tsc_cache/synapse-portal-framework",
+    "noEmit": false,
```

Comment:
With `"composite": true`, `noEmit` must be false, since dependent projects use the output files. For projects where we don't use the TypeScript compiler output as a build artifact, we can redirect the output to a cache folder in node_modules. Idea taken from [this doc from moonrepo](https://moonrepo.dev/docs/config/toolchain#routeoutdirtocache), which we don't use, but they have a [fantastic document](https://moonrepo.dev/docs/guides/javascript/typescript-project-refs) that thoroughly explains setting up a TypeScript project references in a monorepo.